ui/components/DataTable.tsx

```diff
@@ -161,31 +161,28 @@ function filterText(
 
   return _.filter(rows, (row) => {
     let matches = false;
-    for (const colName in row) {
-      const value = row[colName];
 
-      const field = _.find(fields, (f) => {
-        if (typeof f.value === "string") {
-          return f.value === colName;
-        }
-
-        if (f.sortValue) {
-          return f.sortValue(row) === value;
-        }
-      });
+    fields.forEach((field) => {
```

Comment:
I think this needs to be simplified further. Let's say that both name and namespace (in that order) is text searchable. If you search for 1 text filter that matches the name, then I think the namespace would set matches back to false. So I think the matches filter needs to be more carefully passed from "layer to layer", instead of just building up one value as we loop.

The text filters seem like they should be ANDed based on the tests, so I'm going to assume that's correct.

To make this work properly, I think we need to:
 * First (inner-most) OR each field together - any field that matches a single text filter means that the whole row matches that filter.
 * Then, AND the filters together.
 * And only then do we decide if the row as a whole matches the filters as a whole.

We're currently merging the filters before we're merging the columns, which I think doesn't work properly with multiple textSearchable columns - if again name and namespace are searchable, you wouldn't get a hit if one filter matches name and one filter matches namespace, only if all filters matches either name or namespace.

So I would change this whole loop to:

```
  return rows.filter((row) => {
    return textFilters.every((filter) => {
      return fields.some((field) => {
        if (!field.textSearchable) return false;

        let value;
        if (field.sortValue) {
          value = field.sortValue(row);
        } else if (typeof field.value === "function") {
          value = field.value(row);
        } else {
          value = row[field.value];
        }

        return value.includes(filter);
      });
    });
  });
```

Perhaps we could even turn that `value` extraction into some helper function. 

I like chaining together `every` and `some` because they help separate "checking the condition" from "combining the outputs" so they take care of the `matches` malarkey for us - we only return a boolean, and the right thing comes out, and they even handles short-circuiting for us.

Reply:
I don't think it was ever intended to text search any column other than name - I took a look through and no tables have more than one column searchable. I'm down to add this because why not - but to me it'd be weird to have something be filterable AND text searchable.

Reply:
Should I add textSearchable on all Namespace and Cluster columns? Any other columns that might be useful?